controller/api/public/stat_summary.go

```diff
@@ -321,12 +478,20 @@ func buildRequestLabels(req *pb.StatSummaryRequest) (labels model.LabelSet, labe
 
 	default:
 		labelNames = promGroupByLabelNames(req.Selector.Resource)
-
 		labels = labels.Merge(promQueryLabels(req.Selector.Resource))
-		labels = labels.Merge(promDirectionLabels("inbound"))
+
+		if req.Selector.Resource.Type == k8s.TrafficSplit {
+			labels = labels.Merge(promQueryLabels(req.Selector.Resource))
+			labels = labels.Merge(promDirectionLabels("outbound"))
+		} else {
+			labels = labels.Merge(promDirectionLabels("inbound"))
+		}
 	}
 
-	return
+	if req.Selector.Resource.Type == k8s.TrafficSplit {
```

Comment:
I think this overloads the behavior of `buildRequestLabels` too much.  What about creating a `buildTrafficSplitRequestLabels` function that does what we want directly?